src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -251,51 +252,51 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig, protected
-    executeGoogleRequestAsync(googleProject, clusterName, request)
+  private def getCluster(googleProject: GoogleProject, clusterName: ClusterName)(implicit executionContext: ExecutionContext): Future[GoogleCluster] = {
+    val request = dataproc.projects().regions().clusters().get(googleProject.string, dataprocConfig.dataprocDefaultRegion, clusterName.string)
+    executeGoogleRequestAsync(googleProject, "Cluster " + clusterName.string, request)
   }
 
   /**
     * Gets a compute Instance from the API.
     */
-  private def getInstance(googleProject: GoogleProject, zone: String, name: String)(implicit executionContext: ExecutionContext): Future[Instance] = {
-    val request = compute.instances().get(googleProject, zone, name)
-    executeGoogleRequestAsync(googleProject, name, request)
+  private def getInstance(googleProject: GoogleProject, zone: Zone, instanceName: InstanceName)(implicit executionContext: ExecutionContext): Future[Instance] = {
+    val request = compute.instances().get(googleProject.string, zone.string, instanceName.string)
+    executeGoogleRequestAsync(googleProject, "Instance " + instanceName.string, request)
   }
 
   /**
     * Gets an Operation from the API.
     */
-  private def getOperation(operationName: String)(implicit executionContext: ExecutionContext): Future[DataprocOperation] = {
-    val request = dataproc.projects().regions().operations().get(operationName)
-    executeGoogleRequestAsync("operation", operationName, request)
+  private def getOperation(operationName: OperationName)(implicit executionContext: ExecutionContext): Future[DataprocOperation] = {
+    val request = dataproc.projects().regions().operations().get(operationName.string)
+    executeGoogleRequestAsync(GoogleProject("operation"), "Operation " + operationName.string, request)
   }
 
   /**
     * Gets the master instance name from a dataproc cluster, with error handling.
     * @param cluster the Google dataproc cluster
     * @return error or master instance name
     */
-  private def getMasterInstanceName(cluster: GoogleCluster): Option[String] = {
+  private def getMasterInstanceName(cluster: GoogleCluster): Option[InstanceName] = {
     for {
       config <- Option(cluster.getConfig)
       masterConfig <- Option(config.getMasterConfig)
       instanceNames <- Option(masterConfig.getInstanceNames)
       masterInstance <- instanceNames.asScala.headOption
-    } yield masterInstance
+    } yield InstanceName(masterInstance)
   }
 
   /**
     * Gets the zone (not to be confused with region) of a dataproc cluster, with error handling.
     * @param cluster the Google dataproc cluster
     * @return error or the master instance zone
     */
-  private def getZone(cluster: GoogleCluster): Option[String] = {
-    def parseZone(zoneUri: String): String = {
+  private def getZone(cluster: GoogleCluster): Option[Zone] = {
+    def parseZone(zoneUri: String): Zone = {
```

Comment:
...then again, if we kept the `Zone` value class we could put this awful one-off `parseZone` function I made in `Zone.apply(String)`. :)